### PR TITLE
Docs for API Server

### PR DESCRIPTION
**Pull request by MattiasGees**
**What this PR does / why we need it**:
This will add basic docs for API Server

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/user-guide.rst [hunk 1]
```diff
@@ -413,6 +413,19 @@ configuration like that:
       enabled: true
       externalScrapeTargetsOnly: true
 
+
+API Server
+~~~~~~~~~~~
+
+It is possible to let Tarmak create an public endpoint for your APIserver.
+This can be used together with `Secure public endpoints <user-guide.rst#secure-public-endpoints>`_.
```
Comment (dippynark) [inline docs/user-guide.rst]: This link isn't going to work on the tarmak docs website AFAIK (and doesn't work for me locally either)

File: docs/user-guide.rst [hunk 2]
```diff
@@ -413,6 +413,19 @@ configuration like that:
...
+
+.. code-block:: yaml
+
+  kubernetes:
+    apiServer:
+        public: true
```
Comment (dippynark) [inline docs/user-guide.rst]: fix indentation

File: docs/user-guide.rst [hunk 3]
```diff
@@ -465,7 +478,10 @@ API Server
 ++++++++++
 
 For API server you can overwrite the environment level by adding ``allowCIDRs``
-to the kubernetes block
+to the kubernetes block.
+
+.. warning::
+  For this to work, you need to set your `API Server public <user-guide.rst#api-server>`_ first.
```
Comment (dippynark) [inline docs/user-guide.rst]: same as above

**Comment by simonswine** [comment]
Thanks 

/approve

**Pull request merged**